app/services/holiday_factory.rb

```diff
@@ -17,7 +17,51 @@ def create
       holiday_start_date = term.end_date + 1.day
       holiday_end_date = next_term.start_date - 1.day
 
-      CalendarEvent.where(calendar: @calendar, calendar_event_type: holiday_type, start_date: holiday_start_date, end_date: holiday_end_date).first_or_create
+      if holiday_end_date >= holiday_start_date
+        CalendarEvent.where(calendar: @calendar, calendar_event_type: holiday_type, start_date: holiday_start_date, end_date: holiday_end_date).first_or_create!
+      end
     end
   end
+
+  def with_neighbour_updates(calendar_event, attributes)
```

Comment:
Unsurprisingly, Code Climate doesn't like this method....

Reply:
I'm not sure I do either

Reply:
Code Climate never is going to like how *similar* this code is I guess, but maybe extracting some methods may help?

Reply:
I briefly thought about what I could do. The trouble is the code is structurally similar but one almost does the opposite of the other (`start_date` v `end_date`, `>` vs `<` etc). I'll have another look at it and see where I get.

Reply:
I've pushed some changes that reduce the complexity via private methods. I'd be hard pushed to get it much further down.